src/cargo/util/config.rs

```diff
@@ -31,11 +31,7 @@ pub struct Config {
 }
 
 impl Config {
-    pub fn new(shell: MultiShell) -> CargoResult<Config> {
-        let cwd = try!(env::current_dir().chain_error(|| {
```

Comment:
Ah ok, makes sense to me!

In that case it may also be worth moving out the calculation of the `homedir` at some point, but that can happen at a later date